FWCore/Framework/src/EventProcessor.cc

```diff
@@ -1155,7 +1158,9 @@ namespace edm {
                                                           [this,i,h = holder](std::exception_ptr const* iPtr) mutable
                   {
                     if(iPtr) {
-                      h.doneWaiting(*iPtr);
+                      WaitingTaskHolder tmp(h);
```

Comment:
For my own notes: making the copy avoids having `dineWaiting` launch the task before the call to `streamEndLumiAsync`.